### PR TITLE
Foundry integration

### PR DESCRIPTION
**Pull request by bluele**
I propose the integration of [Foundry](https://github.com/foundry-rs/foundry) into the project.

In particular, Forge contained in Foundry enables precise testing of contracts (including unit test) and analysis of gas costs.

### CODE REVIEW

File: tests/foundry/src/IBC.t.sol [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    /* test cases */
+
+    function testClientRegistration() public view {
+        (address client, bool found) = host.getClientImpl(mockClientType);
+        assert(found);
+        assert(address(mockClient) == client);
+    }
+
+    function testSendPacket() public {
+        Packet.Data memory packet = Packet.Data({
```
Comment (siburu) [inline tests/foundry/src/IBC.t.sol:112]: Why don't you use `getPacket()` to make an instance of `Packet.Data`?

File: tests/foundry/src/IBC.t.sol [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            source_port: portId,
+            source_channel: "channel-0",
+            destination_port: portId,
+            destination_channel: "channel-0",
+            data: bytes("{\"amount\": \"100\"}"),
+            timeout_height: Height.Data({revision_number: 0, revision_height: 100}),
+            timeout_timestamp: 0
+        });
+
+        vm.expectEmit(true, false, false, true);
```
Comment (siburu) [inline tests/foundry/src/IBC.t.sol:139]: Is `vm.expectEmit(false, false, false, true)` more suitable to your intension?

File: tests/foundry/src/IBC.t.sol (hunk 1 above)
Comment (bluele) [reply tests/foundry/src/IBC.t.sol]: fixed at https://github.com/hyperledger-labs/yui-ibc-solidity/pull/92/commits/d8c164a8a2862fd4732a45d744a12b30477287a7

File: tests/foundry/src/IBC.t.sol (hunk 2 above)
Comment (bluele) [reply tests/foundry/src/IBC.t.sol]: good catch, I fixed it.

**Comment by bluele** [comment]
hey @siburu, I've fixed and could you review again?

**Review by siburu — APPROVED** [review]
LGTM!

**Comment by bluele** [comment]
It seems that merging this repository requires to pass a job with `name: Test`. I pushed [a fix](https://github.com/hyperledger-labs/yui-ibc-solidity/pull/92/commits/fadd321840b9c32b32b3712493ce05cb3cebed2f) for reverting the name and now I can merge.

For the time being, we will merge this PR by reverting the job name back to the original. However, I think it would be very useful if the committers could configure this kind of settings. @ryjones

**Pull request merged**